x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/indices/recovery/SearchableSnapshotRecoveryState.java

```diff
@@ -37,7 +37,7 @@ public synchronized RecoveryState setStage(Stage stage) {
         return super.setStage(stage);
     }
 
-    public synchronized void preWarmFinished() {
+    public synchronized void setPreWarmFinished() {
```

Comment:
Naming nit, I think we should call this `preWarmComplete` (here and related names) rather than `preWarmFinished` since if there are any failures the prewarming process still finishes, although the shard was not completely prewarmed.

```suggestion
    public synchronized void setPreWarmComplete() {
```